metrics/web/docs/02_presentation_layer_architecture.md

```diff
@@ -21,30 +34,93 @@
+3. Low-level widgets - the configurable widgets that simply displays the data. They should accept the Dart's native types, like `int`, `String`, `Point`, `bool` (can accept types from any UI packages if necessary). They should be highly reusable. Low-level widgets should be placed under the `common/presentation` package.
 
 > Explain the way of using the `strings` in the widgets and where to place them.
 
+Once we have widgets, we probably have some constant texts in them like titles, descriptions, error messages, etc. To make these strings reusable in different parts of our application like tests or even `ci integrations module`, we should extract strings to the separate classes. Commonly, we have a single class with strings for a module, and it is placed under the `module_name/presentation/strings` folder. Another reason to extract the strings into the separate class is translations. To add translations to our application, we have to wrap each string into `Intl.message` method from the [intl](https://pub.dev/packages/intl) package. So, if strings from our application will be placed into one file per module, it will be easier to integrate the translations, by changing the static fields to static getters. 
+
 > Explain the main principles of creating/editing widgets.
 
-[link to widget structure doc.]()
+As all widgets we create are divided into `low-level` and `high-level`, then we should examine the process of creating these types separately. The below statements are the short description of the widgets creating process. See the [Widget structure organization document](03_widget_structure_organization.md) for more details about creating a new widget.
+
+To create a `high-level` widget, we commonly should use the `low-level` widgets in the implementation and create a `view model` for this widget.
+
+To create a `low-level` widget, we should implement the widget that will be highly-configurable and reusable in different parts of the application.
 
 ## Widget structure
 > Create the class diagram explaining the structure of widgets.
 
-> Explain the package structure of the presentation layer
+Let's consider a class diagram explaining the structure of widgets in the Metrics Web Application: 
+
+![Widget Structure Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/software-platform/monorepo/presentation_layer_structure_document/metrics/web/docs/diagrams/widget_structure_class_diagram.puml)
+
+> Explain the package structure of the presentation layer.
+
+The package structure is also an important part of the Metrics Web Application presentation layer. Every presentation unit has a similar package structure. Let's consider the example of the `cool_module`'s package structure.
+
+> * cool_module/
+>    * data/...
+>    * domain/...
+>    * presentation/
+>       * view_model/
```

Comment:
```suggestion
>       * view_models/
```
Should we change it like that? If yes, then we should probably update other documents as well...